oak-blob/src/test/java/org/apache/jackrabbit/oak/spi/blob/split/BlobIdSetTest.java

```diff
@@ -0,0 +1,238 @@
+        String[] blobIds = {"a", "b", "c"};
+        for (String id : blobIds) {
+            blobIdSet.add(id);
+        }
+
+        // Verify the file contains the added blob IDs
+        File storeFile = new File(tempDir, TEST_FILENAME);
+        Assert.assertTrue("Store file should exist", storeFile.exists());
+
+        List<String> fileContent = Files.readAllLines(storeFile.toPath());
+        Assert.assertEquals("File should contain all added blob IDs", blobIds.length, fileContent.size());
+
+        for (int i = 0; i < blobIds.length; i++) {
+            Assert.assertEquals("File line should match blob ID", blobIds[i], fileContent.get(i));
+        }
+    }
+
+    @Test
+    public void testBloomFilterPreventsUnnecessaryFileReads() throws IOException {
+        // The bloom filter should prevent checking the file for non-existent IDs
+
+        // Add some entries to populate the bloom filter
+        for (int i = 0; i < 10; i++) {
+            blobIdSet.add("existing-" + i);
+        }
+
+        // Using a unique pattern for non-existent IDs to ensure they hash differently
+        // than the ones we added
+        for (int i = 0; i < 10; i++) {
+            Assert.assertFalse(blobIdSet.contains("definitely-not-there-" + i));
+        }
+    }
+
+    @Test
+    public void testCachingBehavior() throws IOException {
+        String blobId = "cachedblob";
+
+        // Add the blob ID
+        blobIdSet.add(blobId);
+
+        // First check should populate cache
+        Assert.assertTrue(blobIdSet.contains(blobId));
+
+        // Even if we delete the file, the cached result should be used
+        File storeFile = new File(tempDir, TEST_FILENAME);
+        storeFile.delete();
+
+        // Should still return true due to cache
+        Assert.assertTrue(blobIdSet.contains(blobId));
+    }
+
+    @Test
+    public void testContainsFindsExistingEntriesInFile() throws IOException {
+        // Create some blob IDs
+        String[] blobIds = {"fileblob1", "fileblob2", "fileblob3"};
+
+        // Write blob IDs directly to file (not using BlobIdSet.add())
+        File storeFile = new File(tempDir, TEST_FILENAME);
+        try (FileWriter writer = new FileWriter(storeFile)) {
+            for (String id : blobIds) {
+                writer.write(id + "\n");
+            }
+        }
+
+        // Create a new BlobIdSet instance (which should load from the file)
+        BlobIdSet newBlobIdSet = new BlobIdSet(tempDir.getAbsolutePath(), TEST_FILENAME);
+
+        // Verify that contains() finds all the IDs
+        for (String id : blobIds) {
+            Assert.assertTrue("Should contain blob ID written directly to file: " + id, newBlobIdSet.contains(id));
+        }
+
+        // Verify a non-existent ID still returns false
+        Assert.assertFalse(newBlobIdSet.contains("notinfile"));
+    }
+
+    @Test
+    public void testBloomFilterFalsePositiveProbabilityLessThanThreePercent() throws IOException {
```

Comment:
This test takes around 16 seconds, which in my view is far too long

Reply:
fixed in https://github.com/apache/jackrabbit-oak/pull/2399/commits/c4b1d1349ac6a8ce7b0a4cef69fd85dda20854da